TelCo.ColorCoder/GetNumber.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TelCo.ColorCoder
+{
+    public static class GetNumber
+    {
+        /// <summary>
+        /// Given the two colors the function returns the pair number corresponding to them
+        /// </summary>
+        /// <param name="pair">Color pair with major and minor color</param>
+        /// <returns></returns>
+        public static int GetPairNumberFromColor(ColorPair pair)
+        {
+            // Find the major color in the array and get the index
+            int majorIndex = -1;
+            for (int i = 0; i < Program.colorMapMajor.Length; i++)
+            {
+                if (Program.colorMapMajor[i] == pair.majorColor)
+                {
+                    majorIndex = i;
+                    break;
+                }
+            }
+
+            // Find the minor color in the array and get the index
+            int minorIndex = -1;
```

Comment:
Notice the repeated patterns between major and minor: initialize to -1, iterate and compare. Try moving it to a common function and call in both places, to remove the duplication